test/box-luatest/gh_5529_sort_order_test.lua

```diff
@@ -178,3 +188,237 @@ g1.test_sequential = function(cg)
         t.assert_equals(idx:select(), expect)
     end, {cg.params.engine, cg.params.is_unique})
 end
+
+g_lua_api.test_lua_api = function(cg)
+    cg.server:exec(function()
+        local function is_number(x)
+            return type(x) == 'number'
+        end
+
+        local function key_def_max_indexed_field(key_def)
+            t.assert(key_def ~= nil)
+            t.assert(key_def.parts ~= nil)
+            local result = -1
+            for _, part in pairs(key_def.parts) do
+                if part.fieldno > result then
+                    result = part.fieldno
+                end
+            end
+            return result
+        end
+
+        local function key_def_fill_field_is_indexed_by_part(key_def)
+            local field_is_indexed_by_part = {}
+            for partno, part in pairs(key_def.parts) do
+                -- One field can only be indexed by one part.
+                t.assert(field_is_indexed_by_part[part.fieldno] == nil)
+                field_is_indexed_by_part[part.fieldno] = partno
+            end
+            key_def.field_is_indexed_by_part = field_is_indexed_by_part
+        end
+
+        local function key_def_part_that_indexes_the_field(key_def, fieldno)
+            t.assert(key_def ~= nil)
+            -- The key_def.field_is_indexed_by_part should be filled earlier.
+            t.assert(key_def.field_is_indexed_by_part ~= nil)
+            t.assert(is_number(fieldno))
+            t.assert(fieldno > 0)
+            return key_def.field_is_indexed_by_part[fieldno]
+        end
+
+        -- Generates the values that are possible for this field according to
+        -- the key_def given. The values are returned in an array and sorted
+        -- in ascending order.
```

Comment:
> <...> sorted in ascending order.

As I see, they're sorted according to the sort order in the given key part. Is it correct?

Reply:
Yes, fixed the comment.